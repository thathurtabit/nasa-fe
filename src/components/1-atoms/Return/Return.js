import React from 'react';
import PropTypes from 'prop-types';
import ReturnStyled from './Return.styled';

const Return = ({ text }) => <ReturnStyled to="/">{text}</ReturnStyled>;

export default Return;

Return.propTypes = {
  text: PropTypes.string.isRequired,
};
