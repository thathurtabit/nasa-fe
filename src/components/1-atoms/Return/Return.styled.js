import styled from 'styled-components';
import { Link } from 'react-router-dom';

const ReturnStyled = styled(Link)`
  background: ${props => props.theme.return.bg};
  color: ${props => props.theme.return.text};
  font-size: calc(${props => props.theme.fonts.baseSize} - 3px);
  letter-spacing: 1px;
  padding: 5px 10px;
  text-transform: uppercase;

  &:hover,
  &:focus {
    background: ${props => props.theme.return.hover};
    color: ${props => props.theme.return.text};
    text-decoration: none;
  }
`;

export default ReturnStyled;
