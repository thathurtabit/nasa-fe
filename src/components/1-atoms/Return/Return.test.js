import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import 'jest-styled-components';
import {
  shallowWithTheme,
  mountWithTheme,
} from '../../../utils/test_config/testHelpers';

import Return from './Return';

const mockClickCallback = jest.fn();
const returnText = 'Return title';

describe('Return', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<Return text={returnText} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });

  it('should return the correct text', () => {
    const element = mountWithTheme(
      <BrowserRouter>
        <Return text={returnText} onClick={mockClickCallback} />
      </BrowserRouter>
    );
    expect(element.text()).toBe(returnText);
  });
});
