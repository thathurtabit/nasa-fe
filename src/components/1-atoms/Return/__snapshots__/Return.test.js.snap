// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
